test/backend/declarer_test.dart

```diff
@@ -283,6 +283,22 @@ void main() {
 
       await _runTest(tests.single, shouldFail: true);
     });
+
+    test("runs in the same error zone as the test", () {
+      return expectTestsPass(() {
+        Future future;
+        tearDown(() {
+          // If the tear-down is in a different error zone than the test, the
+          // error will try to cross the zone boundary and get top-leveled.
+          expect(future, throwsA("oh no"));
+        });
+
+        test("test", () {
+          future = new Future.error("oh no");
```

Comment:
Just curiosity: Why do you need to run each test in its own error zone?
Is it to prevent errors leaking between tests? Otherwise, you could potentially run everything in the same error-zone, and then multiplex the handling depending on a zone variable. Each test could still have its own handler, it's just registered on the global test error handler, not on the individual zone.